index.ts

```diff
@@ -0,0 +1,432 @@
+import crypto from "crypto";
+import express, { Request } from "express";
+import { z, ZodError } from "zod";
+
+enum MaterialTypeEnum {
+  ARTICLE = "ARTICLE",
+  VIDEO = "VIDEO",
+  COURSE = "COURSE",
+  IMAGE = "IMAGE",
+  DOCUMENTATION = "DOCUMENTATION",
+  OTHER = "OTHER",
+}
+
+/* SCHEMAS */
+
+// MaterialType
+const materialTypeSchema = z.enum(MaterialTypeEnum);
+
+// Categories
+
+// Category request body
+const categoryDefSchema = z.object({
+  name: z.string(),
+  description: z.string(),
+});
+
+// Category entity in DB
+const categorySchema = categoryDefSchema.extend({
+  id: z.string(),
+  createdAt: z.number(),
+  updatedAt: z.number(),
+});
+
+// Category request body for put
+const categoryUpdateSchema = z.object({
+  name: z.string().optional(),
+  description: z.string().optional(),
+});
+
+// Materials
+
+// Material base schema
+const materialBaseSchema = z.object({
+  url: z.url(),
+  type: materialTypeSchema,
+});
+
+// Material request body
+const materialDefSchema = materialBaseSchema.extend({
+  categoryIds: z.string().array(),
+});
+
+// Material entity in DB
+const materialSchema = materialBaseSchema.extend({
+  id: z.string(),
+  createdAt: z.number(),
+  updatedAt: z.number(),
+});
+
+// Material request body for put
+const materialUpdateSchema = z.object({
+  url: z.url().optional(),
+  type: materialTypeSchema.optional(),
+  categoryIds: z.string().array().optional(),
+});
+
+// MaterialCategory
+const materialCategorySchema = z.object({
+  categoryId: z.string().nullable(),
```

Comment:
The categoryId should not be nullable becuase a materialCategory with only a material ID does not contain any meaningful information.